packages/mc-scripts/src/commands/deployment-previews-push.ts

```diff
@@ -0,0 +1,266 @@
+import chalk from 'chalk';
+import prompts from 'prompts';
+import {
+  processConfig,
+  type CustomApplicationData,
+  type CustomViewData,
+} from '@commercetools-frontend/application-config';
+import type { TCliCommandPushDeploymentPreviewOptions } from '../types';
+import CredentialsStorage from '../utils/credentials-storage';
+import {
+  fetchCustomApplication,
+  updateCustomApplicationDeploymentPreview,
+  createCustomApplicationDeploymentPreview,
+} from '../utils/graphql-requests';
+
+const credentialsStorage = new CredentialsStorage();
+
+const deploymentPreviewAliasRegex = /^[^-#]([a-z]|[-](?![-])){0,62}[^-#]$/g;
+export const fqdnRegex =
+  /(?=^.{4,253}$)(^((?!-)[a-zA-Z0-9-]{0,62}[a-zA-Z0-9]\.)+[a-zA-Z]{2,63}$)/i;
```

Comment:
Wondering if we should move this validation in the API. In the MC we only check this client side. Just to avoid duplication.

Can be a follow up of course.

Reply:
Could you check [this comment](https://github.com/commercetools/merchant-center-application-kit/pull/3541#discussion_r1616261551)?

It's about how to share the validations but also how to adjust them.

Reply:
Yes I saw that. My point was mostly about having this validation additionally in the MC API.

Reply:
I agree we're missing the validation in the backend. We will need to create a follow-up task for that.